### PR TITLE
TYP: read_sas

### PR DESCRIPTION
**Pull request by twoertwein**

### CODE REVIEW

File: pandas/io/sas/sas7bdat.py [hunk 1]
```diff
@@ -744,7 +744,7 @@ def read(self, nrows: int | None = None) -> DataFrame | None:
             raise EmptyDataError("No columns to parse from file")
 
         if nrows > 0 and self._current_row_in_file_index >= self.row_count:
-            return None
+            return pd.DataFrame()
```
Comment (twoertwein) [inline pandas/io/sas/sas7bdat.py:747]: Could also keep None but then read_sas will return `DataFrame | None`.

Comment (jonashaag) [reply pandas/io/sas/sas7bdat.py:747]: I think this breaks `__next__`

**Comment by jonashaag** [comment]
Does this need documentation because we’re breaking the API or do we consider it internal with no guarantees?

**Comment by twoertwein** [comment]
> Does this need documentation because we’re breaking the API or do we consider it internal with no guarantees?

If we want this change @jreback  (I certainly would like it from a typing perspective), I think it should have a whatsnew. Otherwise, I'm happy to keep `None` but then adjust the overloads of `read_sas` to include `None`

**Comment by jreback** [comment]
i think it's fine

just add a note on the very slight api change

**Review by mroeschke — COMMENTED** [review]
LGTM. Off to @jreback

**Review by jreback — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @twoertwein

**Pull request merged**